### PR TITLE
[NNC] added skeleton for refactoring

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56456 Added matmul/unified dtypes
* #55372 [NNC] Move computeNOperand operations over
* **#55371 [NNC] added skeleton for refactoring**

Differential Revision: [D27616418](https://our.internmc.facebook.com/intern/diff/D27616418)

### CODE REVIEW

**Review by ZolotukhinM — APPROVED** [review]
Looks good overall! I left several minor remarks below:

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 1]
```diff
@@ -422,6 +467,26 @@ std::vector<ExprHandle> TensorExprKernel::inferSizesForValue(
...
+    return LongImm::make(*i);
+  } else if (auto b = c10::get_if<bool>(&v)) {
+    return BoolImm::make(*b);
+  // } else if (val->hasNone()) {
+  //   // This is just a placeholder so we don't throw.  None-handling
+  //   // is operator-specific and should be handled properly in
+  //   // the operator-specific lowering code.
+  //   return IntImm::make(0);
+  } else {
+    throw unsupported_dtype();
```
[the remark this answers is not in the document]
Comment (Chillee) [reply torch/csrc/jit/tensorexpr/kernel.cpp:485]: It's the same one that's used right below it (in the original `constant`).

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 2]
```diff
@@ -422,6 +467,26 @@ std::vector<ExprHandle> TensorExprKernel::inferSizesForValue(
...
+  }
+  assert(false);
```
[the remark this answers is not in the document]
Comment (Chillee) [reply torch/csrc/jit/tensorexpr/kernel.cpp:487]: I think I had a lint error at some point ... which is why I did this.

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 3]
```diff
@@ -665,6 +779,30 @@ Tensor* TensorExprKernel::computeTwoOperand(
       });
 }
 
+Tensor* TensorExprKernel::computeTwoOperand(
+    const std::string& name,
+    const std::vector<ArgValue> inputValues,
```
[the remark this answers is not in the document]
Comment (Chillee) [reply torch/csrc/jit/tensorexpr/kernel.cpp:777]: I think I'm going to refrain from this, as I may want to combine these various `computeNOperand` functions in a follow up PR.

**Pull request closed without merging**